### DISCUSSION TITLE
How to run shell command aliases?

### DISCUSSION THREAD

**Discussion by Yevgnen** [helix-editor/helix#3418] [Q&A]
Running shell commands (`:sh`) defined in `.bash_aliases` (which is sourced in `.bashrc`) fails. How can I make helix recognize such commands? Thanks.

**Comment by the-mikedavis** [comment]
You can configure your `editor.shell` to use bash instead of the default `sh` https://docs.helix-editor.com/master/configuration.html#editor-section

**Reply by Yevgnen** [reply]
Hi, I changed it to `shell = ["bash", "-c"]` but it still failed. Am I missing something here?

**Reply by the-mikedavis** [reply]
The bash man page has some additional info: https://linux.die.net/man/1/bash (see the invocation section in particular). It looks like you may be able to set the BASH_ENV environment variable to tell bash where to source for a non-interactive invocation

**Reply by Yevgnen** [reply]
Thanks for the hint. 

I read some SO posts and they point out BASH_ENV too.

I export the variable and check if helix derives it by `! env | grep BASH_ENV` and it does. But still when I do `! ll` (`alias ll=ls -l`), I see a `2022-08-14T22:46:08.301 helix_term::commands [ERROR] Shell error: bash: line 1: ll: command not found` in `~/.cache/helix/helix.log`.

I also try to pass `-I`, but `hx` then crashes when `:sh` without any logs...

**Reply by ggbalaazs** [reply]
It works for me. Defining `BASH_ENV` should be fine, just beware that e.g. in Ubuntu `.bashrc` returns immediately from non-interactive shells. See https://askubuntu.com/questions/224669/how-to-run-some-alias-command-in-bashs-non-interactive-mode

So comment non-interactive return from `.bashrc`, add `export BASH_ENV=~/.bashrc` to it and make sure to put the prompt related section under an `if [[ "$-" == *"i"* ]]; then` to only run them in interactive shells. Also edit helix config `shell = ["bash", "-O", "expand_aliases", "-c"]` or set it in `.bashrc` like `shopt -s expand_aliases`.
